sys/net/gnrc/netif/gnrc_netif_device_type.c

```diff
@@ -0,0 +1,156 @@
+                }
+#endif  /* MODULE_NORDIC_SOFTDEVICE_BLE */
+#if defined(MODULE_CC110X) || defined(MODULE_NRFMIN)
+            case NETDEV_TYPE_CC110X:
+            case NETDEV_TYPE_NRFMIN:
+                if (addr_len <= 3) {
+                    _create_iid_from_short(addr, addr_len, iid);
+                    return sizeof(eui64_t);
+                }
+                else {
+                    return -EINVAL;
+                }
+#endif  /* defined(MODULE_CC110X) || defined(MODULE_NRFMIN) */
+            default:
+                (void)addr;
+                (void)addr_len;
+                (void)iid;
+#ifdef DEVELHELP
+                LOG_ERROR("gnrc_netif: can't convert hardware address to IID "
+                          "on interface %u\n", netif->pid);
+#endif  /* DEVELHELP */
+                assert(false);
+                break;
+        }
+    }
+#endif /* GNRC_NETIF_L2ADDR_MAXLEN > 0 */
+    return -ENOTSUP;
+}
+
+int gnrc_netif_ipv6_iid_to_addr(const gnrc_netif_t *netif, const eui64_t *iid,
+                                uint8_t *addr)
+{
+    assert(netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR);
+    switch (netif->device_type) {
+#if defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW)
+        case NETDEV_TYPE_ETHERNET:
+        case NETDEV_TYPE_ESP_NOW:
+            addr[0] = iid->uint8[0] ^ 0x02;
+            addr[1] = iid->uint8[1];
+            addr[2] = iid->uint8[2];
+            addr[3] = iid->uint8[5];
+            addr[4] = iid->uint8[6];
+            addr[5] = iid->uint8[7];
+            return ETHERNET_ADDR_LEN;
+#endif  /* defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW) */
+#if defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_XBEE)
+        case NETDEV_TYPE_IEEE802154:
+            /* assume address was based on EUI-64
+             * (see https://tools.ietf.org/html/rfc6775#section-5.2) */
+            memcpy(addr, iid, sizeof(eui64_t));
+            addr[0] ^= 0x02;
+            return sizeof(eui64_t);
+#endif  /* defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_XBEE) */
+#ifdef MODULE_NRFMIN
+        case NETDEV_TYPE_NRFMIN:
+            addr[0] = iid->uint8[6];
+            addr[1] = iid->uint8[7];
+            return sizeof(uint16_t);
+#endif  /* MODULE_NETDEV_IEEE802154 */
+#ifdef MODULE_NORDIC_SOFTDEVICE_BLE
+        case NETDEV_TYPE_BLE:
+            memcpy(addr, iid, sizeof(eui64_t));
+            addr[0] ^= 0x02;
+            return sizeof(eui64_t);
+#endif  /* MODULE_NORDIC_SOFTDEVICE_BLE */
+#ifdef MODULE_CC110X
+        case NETDEV_TYPE_CC110X:
+            addr[0] = iid->uint8[7];
+            return sizeof(uint8_t);
+#endif  /* MODULE_CC110X */
+        default:
+            (void)iid;
+            (void)addr;
+#ifdef DEVELHELP
+            LOG_ERROR("gnrc_netif: can't convert IID to hardware address "
+                      "on interface %u\n", netif->pid);
+#endif  /* DEVELHELP */
+            assert(false);
+            break;
+    }
+    return -ENOTSUP;
```

Comment:
Everywhere in this file where an Error-Macro like ENOTSUP is used I get an error about them being undefined.
Not sure if this is because I'm using this against a 2018.10 tree, or a specialty of the esp32 tool chain. Adding an include for errno.h fixes it though.

Reply:
Already fixed in 7ac4b11. :-)